### PR TITLE
Update project information to new name

### LINKED ISSUE #29: Update files to new project name
Many project files, particularly `build.gradle` are not yet updated to _JUnit Pioneer_.

### PR DESCRIPTION
**Pull request by nicolaiparlog**
Closes #29.

--

I hereby agree to the terms of the [Io Contributor License Agreement](../CONTRIBUTING.md#io-contributor-license-agreement).

### CODE REVIEW

**Comment by nicolaiparlog** [comment]
Depending on how #31 turns out, I might have to go back and add the issue numbers to the commit messages.

File: settings.gradle [hunk 1]
```diff
@@ -1,2 +1 @@
-rootProject.name = 'junit-pioneer-parent'
```
Comment (nicolaiparlog) [inline settings.gradle]: I changed this back to leave it to #25 to do the multi-module build in one swoop.

File: build.gradle [hunk 2]
```diff
@@ -106,15 +106,15 @@ publishing {
             pom.withXml {
                 asNode().with {
                     appendNode('packaging', 'jar')
-                    appendNode('name', 'JUnit Io')
+                    appendNode('name', 'JUnit Pioneer')
                     appendNode('description', 'JUnit 5 Extension Pack')
-                    appendNode('url', 'https://github.com/CodeFX-org/junit-io')
+                    appendNode('url', 'https://github.com/junit-pioneer/junit-pioneer')
```
Comment (nicolaiparlog) [inline build.gradle]: Until we actually have a site, I decided to use the GitHub repo.

**Comment by nicolaiparlog** [comment]
This should fix all references to the old name "JUnit Io" except for packages (see #24) and the CI badges in the README (see #41).

@smoyer64 Can you have a look? (You won't believe how many hits you get when doing a full text search for "io". Extension, exception, ...)

File: src/test/java/org/codefx/junit/io/extension/OsConditionSmokeTests.java [hunk 3]
```diff
@@ -12,20 +12,17 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
-import static org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder.request;
 
-import org.codefx.junit.io.AbstractIoTestEngineTests;
+import org.codefx.junit.io.AbstractPioneerTestEngineTests;
```
Comment (smoyer64) [inline src/test/java/org/codefx/junit/io/extension/OsConditionSmokeTests.java]: Io was removed from the class name but the package still needs to be changed.

File: src/test/java/org/codefx/junit/io/vintage/TestIntegrationTests.java [hunk 4]
```diff
@@ -19,14 +19,14 @@
 import java.nio.file.InvalidPathException;
 import java.util.Optional;
 
-import org.codefx.junit.io.AbstractIoTestEngineTests;
+import org.codefx.junit.io.AbstractPioneerTestEngineTests;
```
Comment (smoyer64) [inline src/test/java/org/codefx/junit/io/vintage/TestIntegrationTests.java]: Io was removed from the class name but the package still needs to be changed.

**Comment by nicolaiparlog** [comment]
As mentioned, I leave package names for #24. :wink:

**Comment by smoyer64** [comment]
Wow ... I'm completely out-of-it today (don't trust anything I say or any code I write).

**Comment by nicolaiparlog** [comment]
> (don't trust anything I say or any code I write).

Says it and merges without squashing. 😜 I've done the unthinkable and rewrote history on master, hoping nobody except maybe you would pull it in the 9 minutes it took me.

**Comment by nicolaiparlog** [comment]
The squashed commit is 65a922a, the old merge commit (84c36ef) should disappear soon.

**Comment by smoyer64** [comment]
Sorry about that ... so no more talking, coding or button pushing.  I'd hate to get kicked out of the project!

**Comment by nicolaiparlog** [comment]
No worries, who would code then. :wink:

**Pull request merged**